### PR TITLE
vertex hptuning computer vision nb

### PR DESCRIPTION
**Pull request by kylesteckler**
Solutions notebook for a lab that launches vertex hyperparameter tuning job for image classification use case.

### CODE REVIEW

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,1225 @@
+{
```
Comment (takumiohym) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    train_ds = create_dataset(
</pre><p>If it is just for validation, different variable name may be better (like <code>temp_ds</code>), as it can be confusing to use the same variable that is also used in train.py.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (takumiohym) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    !gcloud builds submit --machine-type=e2-highcpu-32 --timeout=15m --tag $TRAINING_CONTAINER_IMAGE_URI flower_trainer
</pre><p>Just out of curiosity, but do we need to use this large machine to build this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (takumiohym) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #49.    hp_job.run()
</pre><p>maybe add <code>sync=False</code> to run asynchronously? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

[the remark this answers is not in the document]
Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Done.</p>

Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Done.</p>

[the remark this answers is not in the document]
Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>What's the benefit of this? I structured it this way based on the Vertex provided hyperparam tuning colab here: https://codelabs.developers.google.com/vertex_hyperparameter_tuning#3</p>

Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Just to make the build a little faster so there's less waiting in the lab</p>

[the remark this answers is not in the document]
Comment (takumiohym) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>I think the advantage of the Callback function is that we can report metrics in each epochs (not only at the end of the training). So we can check each performance metrics on the console even during each trials.</p>

[the remark this answers is not in the document]
Comment (BenoitDherin) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <ul><li>Learning objective in bold</li></ul>

[the remark this answers is not in the document]
Comment (BenoitDherin) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>train.py does not seem to be pep8 compliant (import order, lines with more than 80 chars, single blank line between function definitions, missing spaces around "=="). Maybe run a pep8 formatter on it. The module URL should probably be defined in a constant, and this constant used as the train_evaluation argument default. </p><p> </p><p>* <code>train_evaluate</code> -&gt; <code>train_and_evaluate</code></p><p>  </p>

[the remark this answers is not in the document]
Comment (BenoitDherin) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>+1</p>

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,1226 @@
...
```
Comment (BenoitDherin) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>* I'd remove the reference to <code>creating_tfrecords_at_scale.ipynb</code> since you have the code to populate the bucket</p><p>* What about moving the data setup cell before trying to access the data?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (sanjanalreddy) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Agreed, the current flow of the lab is a bit non-linear right now. </p>

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb (hunk 1 above)
Comment (sanjanalreddy) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>I had the same question, what's the decrease in training time with using this machine?</p>

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb (hunk 2 above)
Comment (sanjanalreddy) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.    %env TF_CPP_MIN_LOG_LEVEL=3
</pre><p>Level 3 doesn't print the error messages, I would like to see error messages though. So if you want to use this, maybe consider level=1</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (sanjanalreddy) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    BUCKET = PROJECT + "-flowers"
</pre><p>Can we just use the same bucket that we usually use <code>project_id</code> and then create a folder <code>flowers</code>  under this project to keep it uniform with other labs? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sanjanalreddy) [inline notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    FROM gcr.io/deeplearning-platform-release/tf2-gpu.2-7
</pre><p>Consider upgrading to <code>gcr.io/deeplearning-platform-release/tf2-gpu.2-8</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/225/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

**Review by BenoitDherin — APPROVED** [review]

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Done.</p>

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb (hunk 2 above)
Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Changed to =1</p>

File: notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kylesteckler) [reply notebooks/image_models/solutions/vertex_tuning_image_classification.ipynb:1]: <p>Done. </p>
